applications/nrf_desktop/src/modules/Kconfig.caf_settings_loader.default

```diff
@@ -10,32 +10,76 @@ config DESKTOP_SETTINGS_LOADER
 	bool "Settings loader module (CAF)"
 	select CAF_SETTINGS_LOADER
 	select SETTINGS
-	imply NVS
 	imply FLASH
 	imply FLASH_MAP
 	imply FLASH_PAGE_LAYOUT
 	help
 	  nRF Desktop uses settings loader module from Common Application
 	  Framework (CAF).
 
-	  By default, nRF Desktop application uses non-volatile storage backend
-	  and the settings partition is located in internal FLASH.
+	  The settings partition is located in internal non-volatile memory.
+	  By default, nRF Desktop application, depending on the non-volatile
+	  memory technology used by the device, uses different settings
+	  backends.
 
 if DESKTOP_SETTINGS_LOADER
 
+choice DESKTOP_SETTINGS_BACKEND
+	prompt "Settings backend"
+	default DESKTOP_SETTINGS_BACKEND_ZMS if SOC_FLASH_NRF_RRAM
+	default DESKTOP_SETTINGS_BACKEND_NVS
+	help
+	  By default, nRF Desktop application, depending on the non-volatile
+	  memory technology used by the device, uses either the Zephyr Memory
+	  Storage (ZMS) or Non-Volatile Storage (NVS) settings backend.
+	  ZMS is used for the devices with the RRAM non-volatile memory that
+	  do not require explicit erase. Otherwise, the NVS is used.
+
+config DESKTOP_SETTINGS_BACKEND_ZMS
+	bool "Zephyr Memory Storage (ZMS) settings backend"
+	select ZMS
+	help
+	  This option enables the Zephyr Memory Storage (ZMS) as the settings
+	  backend for the nRF Desktop application.
+
+config DESKTOP_SETTINGS_BACKEND_NVS
+	bool "Non-Volatile Storage (NVS) settings backend"
+	select NVS
+	help
+	  This option enables the Non-Volatile Storage (NVS) as the settings
+	  backend for the nRF Desktop application.
+
+endchoice
```

Comment:
why do you need this, just imply in the above, if someone wants to change because of external flash then they can change from SETTINGS_ZMS to SETTINGS_NVS, this choice adds nothing

Reply:
FYI @MarekPieta

Reply:
I agree this could be simplified. We can skip configuring `SETTINGS_BACKEND` below as this is autoselected based on FS.
Then you could remove `DESKTOP_SETTINGS_BACKEND` altogether and simply add 
```
config DESKTOP_SETTINGS_LOADER
   imply ZMS if SOC_FLASH_NRF_RRAM
   imply NVS
```

just check if the end values are correct

Reply:
I am worried it might lead to problems if multiple filesystem support is enabled in configuration. E.g. enabling ZMS for any purpose other than settings would silently change the used settings backend - probably not expected. That's why I suggested to make configuration more explicit here.

Can we keep it?

Reply:
No, the correct backend would be on if `SETTINGS_NVS` or `SETTINGS_ZMS` is used, this configuration option is completely pointless, it just mirrors another one adding nothing

Reply:
> I am worried it might lead to problems if multiple filesystem support is enabled in configuration

You are right that this will be a problem. Settings backend is just a dropdown of what is enabled as FS.
The main problem I see is that Zephyr should handle FS in partition selection (in DTS).
At this point seems Zephyr is not prepared well to handle multiple storage partitions (with various FS used), but I would ignore this at this point and assume that one FS is used in entire system at this point.

Reply:
Keep in mind that FS might be also used "as is" for any purpose (i.e. not as settings backend). The approach currently presented in the PR should ensure correct handling of case with multiple filesystems too. Maybe we can stick with it for now?

Reply:
> Keep in mind that FS might be also used "as is" for any purpose (i.e. not as settings backend).

Then you wouldn't enable `SETTINGS_NVS` or `SETTINGS_ZMS`, I'm not really sure why this discussion is still going on

Reply:
> > Keep in mind that FS might be also used "as is" for any purpose (i.e. not as settings backend).
> 
> Then you wouldn't enable `SETTINGS_NVS` or `SETTINGS_ZMS`, I'm not really sure why this discussion is still going on

You actually would if you have one FS used for settings and another for provisioning. I think fast pair case has two storage partitions.
I expect the problem would be if these are handled using two different FSs. Likely not the use case at the moment.

Note that this should be owners and maintainers call of this code to make the decision. If @MarekPieta believes this is the best path forward, let's keep it as is. The code is not wrong and indeed there is no point in arguing.

Reply:
You might use file system "B"  (for an application-specific purpose) next to the Zephyr settings using a file system "A". No one forbids enabling multiple file systems for various non-volatile memories (even though IIRC we currently do not support multiple settings instances). That's why I suggested to add the application-specific Kconfig option that ensures no side effects here.

Still, if you insist we will remove the application-specific Kconfigs and stick to the configuration that relies only on implies (and has a potential side effect).

Reply:
Applied @nordicjm suggestion to switch to implies as the current application-specific Kconfig would not be accepted.